interfaces/apparmor/template.go

```diff
@@ -279,6 +279,8 @@ var templateCommon = `
   /sys/kernel/mm/transparent_hugepage/hpage_pmd_size r,
   /sys/module/apparmor/parameters/enabled r,
   /{,usr/}lib/ r,
+  /var/lib/snapd/hostfs/usr/share/mime/* r,
+  /etc/gnome/defaults.list r,
```

Comment:
The first line also belongs in another interface but I don't understand why would an application look at `/usr/share/mime` through the hostfs directory. 

The second line definitely belongs in another interface, perhaps the `desktop` interface?

Reply:
Thanks for looking, @zyga 

I'm not sure why hostfs is read either. Perhaps it is a fault in snap-store? I may need to investigate that.

When looking at the apparmor profile, here is what mime pops up.

```
sklein@computer:~/Git/snap-store$ cat  /var/lib/snapd/apparmor/profiles/snap.snap-store.ubuntu-software | grep mime
  /etc/mime.types r,
owner @{HOME}/.local/share/mime/**   r,
/var/lib/snapd/desktop/applications/mimeinfo.cache r,
```
>>The second line definitely belongs in another interface, perhaps the desktop interface?

Oh maybe. Here's what it looks like for gnome now.

```
cat ./interfaces/builtin/desktop.go | grep gnome
# subset of gnome abstraction
/usr/share/gnome/glib*/schemas/{,*}        r,
  interface=org.gnome.SettingsDaemon.MediaKeys
  path=/org/gnome/SettingsDaemon/MediaKeys
  path=/org/gnome/SettingsDaemon/MediaKeys
# applications using libgcr (such as pinentry-gnome3) for secure pin
# https://developer.gnome.org/gcr/unstable/GcrPrompt.html
# https://developer.gnome.org/gcr/unstable/GcrSecretExchange.html
    path=/org/gnome/keyring/Prompter
    interface=org.gnome.keyring.internal.Prompter
    path=/org/gnome/keyring/Prompt/p[0-9]*
    interface=org.gnome.keyring.internal.Prompter.Callback
# gnome/kde/freedesktop.org
    path="/{,org/freedesktop/,org/gnome/}ScreenSaver"
    interface="org.{freedesktop,gnome}.ScreenSaver"
    path="/{,org/freedesktop/,org/gnome/}ScreenSaver"
    interface="org.{freedesktop,gnome}.ScreenSaver"
```

Reply:
Sorry, opened a new PR. I'm not 100% on how to change a current PR based on reviews.

See #8609.